File: app/src/main/java/com/aspark/carebuddy/view/MainActivity.java
package com.aspark.carebuddy.view;

import androidx.appcompat.app.AppCompatActivity;

import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import com.aspark.carebuddy.Contract;
import com.aspark.carebuddy.R;
import com.aspark.carebuddy.model.UserModel;
import com.aspark.carebuddy.presenter.LoginPresenter;
import com.aspark.carebuddy.presenter.UserLoginPresenter;
import com.google.android.material.card.MaterialCardView;

public class MainActivity extends AppCompatActivity implements Contract.View{

    MaterialCardView cardView_nurse, cardView_user;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        SharedPreferences preferences = getSharedPreferences(getPackageName(),MODE_PRIVATE);
        boolean isSignedIn = preferences.getBoolean("isSignedIn", false);
        String username = preferences.getString("username",null);

        //TODO to get all the user data send username to backend and get user object in return
        Log.d("MainActivity", "onCreate: currentUser: "+username);
        Log.i("MainActivity", "check isSignedIn "+isSignedIn);

        if (isSignedIn) {

            UserModel.getCurrentUser().setEmail(username);
            Contract.Presenter.PresenterUserLogin presenterUserLogin = new UserLoginPresenter();
            presenterUserLogin.userSignedInLogin(this);
        }

        setContentView(R.layout.activity_main);

        cardView_nurse = findViewById(R.id.nurseCard);
        cardView_user = findViewById(R.id.userCard);

        cardView_nurse.setOnClickListener(view -> {

            Contract.Presenter presenter = new LoginPresenter();
            presenter.nurseCardClickListener(this);
        });

        cardView_user.setOnClickListener(view -> {

            Contract.Presenter presenter = new LoginPresenter();
            presenter.userCardClickListener(this);
        });
    }

    @Override
    public void destroyActivity() {

        finish();
        Log.i("MainActivity", "destroyActivity: Activity Destroyed!");
    }
}

File: app/src/main/java/com/aspark/carebuddy/view/MainActivity.kt
package com.aspark.carebuddy.view

import android.content.Intent
import android.os.Bundle
import android.util.Log
import androidx.activity.viewModels
import androidx.appcompat.app.AppCompatActivity
import com.aspark.carebuddy.MainViewModel
import com.aspark.carebuddy.databinding.ActivityMainBinding
import com.aspark.carebuddy.login.NurseLoginActivity
import com.aspark.carebuddy.login.UserLoginActivity
import com.aspark.carebuddy.model.Nurse.Companion.currentNurse
import com.aspark.carebuddy.model.User
import com.aspark.carebuddy.view.nurse.NurseHomeActivity
import com.aspark.carebuddy.view.user.UserHomeActivity

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private val viewModel: MainViewModel by viewModels()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val preferences = getSharedPreferences(packageName, MODE_PRIVATE)
        val isUserSignedIn = preferences.getBoolean("isUserSignedIn", false)
        val isNurseSignedIn = preferences.getBoolean("isNurseSignedIn", false)

        //TODO to get all the user data send userEmail to backend and get user object in return

        if (isUserSignedIn) {

            val userEmail = preferences.getString("userEmail", null)
            Log.d("MainActivity", "onCreate: currentUser: $userEmail")

            User.currentUser.email = userEmail
            viewModel.callUserHomeActivity()

        } else if (isNurseSignedIn) {

            val nurseEmail = preferences.getString("nurseEmail", null)
            Log.d("MainActivity", "onCreate: currentNurse: $nurseEmail")
            currentNurse.email = nurseEmail!!
            viewModel.callNurseHomeActivity()
        }

        viewModel.startUserHomeActivity.observe(this){

            it?.let {
                val intent = Intent(this,UserHomeActivity::class.java)
                startActivity(intent)
            }
        }

        viewModel.startNurseHomeActivity.observe(this){

            it?.let {
                val intent = Intent(this,NurseHomeActivity::class.java)
                startActivity(intent)
            }
        }

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.nurseCard.setOnClickListener {
            viewModel.callNurseLoginActivity()
        }

        binding.userCard.setOnClickListener {
            viewModel.callUserLoginActivity()
        }

        viewModel.startNurseLoginActivity.observe(this){

            it?.let {
                val intent = Intent(this,NurseLoginActivity::class.java)
                startActivity(intent)
            }
        }

        viewModel.startUserLoginActivity.observe(this){

            it?.let {
                val intent = Intent(this,UserLoginActivity::class.java)
                startActivity(intent)
            }
        }

    }

//    override fun destroyActivity() {
//        finish()
//        Log.i("MainActivity", "destroyActivity: Activity Destroyed!")
//    }
}